rocketpy/prints/rocket_prints.py

```diff
@@ -114,9 +114,13 @@ def rocket_aerodynamics_quantities(self):
         print("\nAerodynamics Lift Coefficient Derivatives\n")
         for surface, position in self.rocket.aerodynamic_surfaces:
             name = surface.name
+            # ref_factor corrects lift for different reference areas
+            ref_factor = (surface.rocket_radius / self.rocket.radius) ** 2
```

Comment:
Defining this property (`ref_factor`) inside the prints module is dangerous.
I recommend defining this directly inside the Rocket class and using the already calculated value here instead of making this multiplication.

There should be no "business logic" within the plots or prints modules.

What do you think?

Reply:
Makes sense! I'll look into a reasonable way to implement this.

Currently, what I have in mind is that each aero surface has three main relevant information when it comes to the Rocket class: itself, it's position along the rocket, and it's reference area factor. Therefore, I will try to implement this alongside the position info.

Reply:
Tooke a quick look and couldn't find a way to integrate it into the `Components` class without making things look disorganized, since it is also used for other componentes which are not aerodynamic.

Two options:
 1. Find another way to implement this.
 2. Create an `AeroComponents` class the inherets from the `Components` class and adds the reference area factor attribute.

What do you think @Gui-FernandesBR?

Reply:
Creating a new class just for this seems a bit overkill to me, at least at a first glance.

I would say that perhaps we could save the ref_factor directly in the AeroSurface object itself. Can't we?

Also, would it be a problem if we had the ref_factor in the component class but leave it equals to 1 by default?

Reply:
> Creating a new class just for this seems a bit overkill to me, at least at a first glance.

I agree.

> I would say that perhaps we could save the ref_factor directly in the AeroSurface object itself. Can't we?

We can, but I strongly advise aignst it. Stand focused a lot on making each aero surface as independent from the Rocket as possible. This is important in the architecture that these components can be moved from rocket to rocket without changing the component itself.

Now, the "local" rocket radius was an information that we needed to have in orther to calculate things like interference factor. Some might argue that the local radius is indeed part of the component structure, so it makes sense to have that info.

Having said that, addind a reference to the "maximum" rocket radius to each component crosses the line in my point of view. Each component would end up having an excessive relationship to the rocket.

> Also, would it be a problem if we had the ref_factor in the component class but leave it equals to 1 by default?

This seems like a suboptimal solution for me. In some cases, this attribute will make no sense.

To be honest, a simple division in the print class doesn't seem like such a bad compromise.

Reply:
> This seems like a suboptimal solution for me. In some cases, this attribute will make no sense.

Can you give me some examples?

> To be honest, a simple division in the print class doesn't seem like such a bad compromise.

Given the discussion, I agree with you.

Reply:
> Can you give me some examples?

For the rail button component.

Reply:
Ok, I see it.

After thinking, thinking, and thinking during these past days.
I know think that the AeroComponent class might be a good idea.
Only problem is that the rail buttons should also be treated as a "aero surface", but anyway.

We should have Component (motors, buttons) and AeroComponents (tails, noses, etc.) being added to the rocket.

The AeroComponent could be related to the Component class via inheritance.

I think we could do that in a separate PR, since the current solution is already working and does not make anything wrong. Also, refactoring this should be easy in the future.

Reply:
I read all the thread and I agree that keeping it as it is seems the best for this hotfix.

In my opinion, the rocket class should be reponsible for storing and mapping each specific surface to their connection parameters (e.g. position, local radius etc).

When dealing with `AeroSurface` we should rely more on its abtraction (base class) than on particular implementations. For instance, about the particular issue raised about `RailButtons`, I believe this is a case that does not matter too much whether the "local radius" has physical meaning: the code should treat it internally as any surface but this parameter would not be exposed to the user.

As a side note, the bigger difference I see when trying to think about an abstraction for surfaces is that some of them are attached to the tube (such as fins) and others goes transversely (like nosecones), which make them quite different geometricaly.

Reply:
I added a new issue to the repo so we can follow up this question later: https://github.com/RocketPy-Team/RocketPy/issues/561

Marking this as solved for now.